lib/Emitter.php

```diff
@@ -25,32 +26,32 @@
             /**
              * @var callable
              */
-            private $resolve;
+            private $complete;
 
             /**
              * @var callable
              */
             private $fail;
 
             public function __construct() {
-                $this->stream = new Internal\PrivateStream(
-                    function (callable $emit, callable $resolve, callable $fail) {
+                $this->iterator = new Internal\PrivateIterator(
+                    function (callable $emit, callable $complete, callable $fail) {
                         $this->emit = $emit;
-                        $this->resolve = $resolve;
+                        $this->complete = $complete;
                         $this->fail = $fail;
                     }
                 );
             }
 
             /**
-             * @return \Amp\Stream
+             * @return \Amp\Iterator
              */
-            public function stream(): Stream {
-                return $this->stream;
+            public function getIterator(): Iterator {
```

Comment:
The same way you could argue that `getIterator` shall return _an_ Iterator from the beginning.
At least that's what `IteratorAggregate::getIterator()` does - it normally returns you a new iterator instance.
Retaining that symmetry there may let you have false expectations.